### PR TITLE
Embeddings Refresher

### PR DESCRIPTION
**Pull request by calumjs**
Adding an Azure Function which will grab rule markdown file changes in the specified time period, embed and upsert to the vector database.

Temporarily deploying as HTTP trigger so I can test it a bit, once happy will convert to timer trigger and forget about it.

Related to #48

### CODE REVIEW

**Review by bradystroud — CHANGES_REQUESTED** [review]
Just the one comment, looks good otherwise

File: .github/workflows/template-build-deploy.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
           subscriptionId: ${{ secrets.AZURE_SUBSCRIPTION }}
           resourceGroupName: ${{ secrets.AZURE_RG }}
           template: ./.github/bicep/main.bicep
-          parameters: 'appName=${{ inputs.appName }} environment=${{ inputs.deployEnvironment }} hostingPlanName=${{ vars.HOSTING_PLAN_NAME }} hostingPlanRgName=${{ vars.HOSTING_PLAN_RG_NAME }} connectionString=${{ secrets.ConnectionString }} openAiApiKey=${{ secrets.OpenAiApiKey }} allowedCors=${{ vars.ALLOWED_CORS }} maxRequests=${{ vars.MAX_REQUESTS }} signingAuthority=${{ vars.SIGNING_AUTHORITY }}'
+          parameters: 'appName=${{ inputs.appName }} environment=${{ inputs.deployEnvironment }} hostingPlanName=${{ vars.HOSTING_PLAN_NAME }} hostingPlanRgName=${{ vars.HOSTING_PLAN_RG_NAME }} connectionString=${{ secrets.ConnectionString }} openAiApiKey=${{ secrets.OpenAiApiKey }} githubToken=${{ secrets.GithubPAT }} allowedCors=${{ vars.ALLOWED_CORS }} maxRequests=${{ vars.MAX_REQUESTS }} signingAuthority=${{ vars.SIGNING_AUTHORITY }}'
```
[the remark this answers is not in the document]
Comment (calumjs) [reply .github/workflows/template-build-deploy.yml]: Done - renamed

**Review by bradystroud — APPROVED** [review]
LGTM 🛶

File: .github/bicep/main.bicep [hunk 2]
```diff
@@ -44,12 +48,17 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2022-05-01' = {
...
+  parent: storageAccount
+  name: 'default'
+}
+
 resource hostingPlan 'Microsoft.Web/serverfarms@2021-03-01' existing = {
   name: hostingPlanName
   scope: resourceGroup(hostingPlanRgName)
 }
 
-resource kv 'Microsoft.KeyVault/vaults@2023-07-01' = {
```
Comment (MattParkerDev) [inline .github/bicep/main.bicep:52]: why was this KV version changed to use an older version? @calumjs

Comment (calumjs) [reply .github/bicep/main.bicep:52]: image
When I tried to build it locally during testing I got a bunch of warnings, so I just swapped it to a value from another pipeline.
Did I miss something?

Comment (calumjs) [reply .github/bicep/main.bicep:52]: Ah ok I see - I just had an old version of bicep installed - no warning after upgrade.
Will put it back to the 2023 version

**Pull request merged**